### PR TITLE
Space Travellers' Hub Project

### PR DESCRIPTION
**Pull request by shaqdeff**
- Setup a react project and installed all required packages and dependencies.
- Setup navigation bar and routes.
- Setup components for each route.
- Created a store to manage the state of applications.
- Created reducers for each slice and added them to the store.
- Implemented and dispatched actions for various features.
- Styled the wireframe.
- Ran tests.

Kindly review and provide feedback. 
Regards.

### CODE REVIEW

File: src/components/Rockets.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/* eslint-disable quotes */
+/* eslint-disable arrow-body-style */
+/* eslint-disable react/jsx-closing-bracket-location */
```
Comment (zieeco) [inline src/components/Rockets.jsx]: - Please, try and write clean codes that conform to the linter. this linter could be fixed manually. please fix this to make your codebase more clean and professional.. Here's the working code::


```
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import Button from 'react-bootstrap/Button';
import ListGroup from 'react-bootstrap/ListGroup';
import Badge from 'react-bootstrap/Badge';
import Image from 'react-bootstrap/Image';
import {
  selectAllRockets,
  fetchRockets,
  reserveRockets,
  cancelRockets,
} from '../redux/rockets/rocketSlice';
import './Rockets.css';

function Rockets() {
  const dispatch = useDispatch();
  const { data: rockets, status } = useSelector(selectAllRockets);

  const handleReserveButton = (id) => {
    dispatch(reserveRockets(id));
  };

  const handleCancelButton = (id) => {
    dispatch(cancelRockets(id));
  };

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchRockets());
    }
  }, [status, dispatch]);

  return (
    <ListGroup className="rocket__container" variant="flush">
      {rockets.map((rocket) => (
        <ListGroup.Item key={rocket.id} className="mt-5 py-4 d-flex">
          <Image
            src={rocket.flickr_images[0]}
            alt="rocket-pic"
            className="rocket__image"
          />
          <div className="mx-4">
            {rocket.reserved ? (
              <Badge
                bg="success"
                style={{
                  fontSize: '1.15rem',
                  padding: '0.5rem 1rem',
                  marginBottom: '0.5rem',
                }}
              >
                Reserved
              </Badge>
            ) : (
              ''
            )}
            <h2 className="h1">{rocket.rocket_name}</h2>
            <p
              className="mt-4 lead"
              style={{ fontSize: '1.4rem', fontWeight: '400' }}
            >
              {rocket.description}
            </p>

            {rocket.reserved ? (
              <Button
                variant="outline-danger btn-lg"
                onClick={() => handleCancelButton(rocket.id)}
                style={{
                  fontWeight: '600',
                  fontSize: '1.3rem',
                  padding: '0.75rem 1rem',
                }}
              >
                Cancel Reservation
              </Button>
            ) : (
              <Button
                className="btn-lg"
                onClick={() => handleReserveButton(rocket.id)}
                style={{
                  fontWeight: '600',
                  fontSize: '1.3rem',
                  padding: '0.75rem 1rem',
                }}
              >
                Reserve Rocket
              </Button>
            )}
          </div>
        </ListGroup.Item>
      ))}
    </ListGroup>
  );
}

export default Rockets;

```

File: src/components/MyProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/* eslint-disable object-curly-newline */
+/* eslint-disable arrow-body-style */
+/* eslint-disable react/jsx-closing-bracket-location */
```
Comment (zieeco) [inline src/components/MyProfile.jsx]: - Please, try and write clean codes that conform to the linter. this linter could be fixed manually. please fix this to make your codebase more clean and professional.. Here's the working code::


- Here's another fix..
```

import {
  Container, Row, Col, ListGroup
} from 'react-bootstrap';
import { useSelector } from 'react-redux';

function MyProfile() {
  // eslint-disable-next-line max-len
  const reservedMissions = useSelector((state) => state.missions.data.filter((mission) => mission.reserved));
  // eslint-disable-next-line max-len
  const reservedRockets = useSelector((state) => state.rockets.data.filter((rocket) => rocket.reserved));

  return (
    <div>
      <Container style={{ margin: '13rem auto 0 auto' }}>
        <Row>
          <Col className="mx-5">
            <h2
              className="text-muted"
              style={{ fontSize: '2.5rem', fontWeight: '600' }}
            >
              My Missions
            </h2>
            <ListGroup
              className="pt-3"
              as="ol"
              numbered
              style={{ fontSize: '1.7rem', fontWeight: '500' }}
            >
              {reservedMissions.map((mission) => (
                <ListGroup.Item
                  className="p-3"
                  as="li"
                  key={mission.mission_id}
                >
                  {mission.mission_name}
                </ListGroup.Item>
              ))}
            </ListGroup>
          </Col>
          <Col className="ml-3">
            <h2
              className="text-muted"
              style={{ fontSize: '2.5rem', fontWeight: '600' }}
            >
              My Rockets
            </h2>
            <ListGroup
              className="pt-3"
              as="ol"
              numbered
              style={{ fontSize: '1.7rem', fontWeight: '500' }}
            >
              {reservedRockets.map((rocket) => (
                <ListGroup.Item className="p-3" key={rocket.id}>
                  {rocket.rocket_name}
                </ListGroup.Item>
              ))}
            </ListGroup>
          </Col>
        </Row>
      </Container>
    </div>
  );
}

export default MyProfile;
```

File: src/components/Missions.jsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
+/* eslint-disable camelcase */
```
Comment (zieeco) [inline src/components/Missions.jsx]: - Please, fix this linter also:
- remove all and look for this `<th />` and remove it

File: src/components/Missions.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(fetchMissions());
+    }
+  }, []);
+
+  /* eslint-disable jsx-a11y/control-has-associated-label */
+  /* eslint-disable react/jsx-closing-bracket-location */
```
Comment (zieeco) [inline src/components/Missions.jsx:21]: - Please, fix this linter also

File: README.md [hunk 5]
```diff
@@ -1,2 +1,74 @@
...
+- Once the files are on your machine, open the **Micro-Space-Hub** folder in [Visual Studio Code](https://code.visualstudio.com/download).
+
+## Deployment
+
+- After opening the files in Visual Studio Code, open the **VS Code** integrated terminal and run the following commands:
+
+```
+npm install
+```
+
```
Comment (zieeco) [inline README.md:42]: - I can install but i can not run your test. Please add an instruction on how to run the test.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,74 @@
...
+👤 **Oluwatomisin Bello**
+
+- GitHub: [@t-bello](https://github.com/t-bello)
+- Twitter: [@__tbello](https://twitter.com/__tbello)
+- LinkedIn: [Oluwatomisin Bello](https://www.linkedin.com/in/tbello7)
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
```
Comment (zieeco) [inline README.md]: - Please, add the link to the issues page of your project to make your readme more professional.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,74 @@
...
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
```
Comment (zieeco) [inline README.md:85]: - Please, kindly add a License file to make your readme more professional.

File: src/tests/Components/__snapshots__/Missions.test.js.snap [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            Status
+          </th>
+          <th />
+        </tr>
+      </thead>
+      <tbody />
+    </table>
+  </div>
+</div>
+`;
```
Comment (zieeco) [inline src/tests/Components/__snapshots__/Missions.test.js.snap]: - Please, note that after fixing the linter issues, you're test will fail. kindly fix this also

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Missions.jsx (hunk 3 above)
Comment (shaqdeff) [reply src/components/Missions.jsx]: I've fixed everything else, but kindly note that, that empty  ```<th />``` is mandatory for React to render the component. And this is why taking it out is breaking the app, and failing the tests. Thank you.

**Pull request merged**